### PR TITLE
Add adapter for opsgenie schedules

### PR DESCRIPTION
**Pull request by bnjns**
Adds an adapter which lets you sync an Opsgenie Schedule. Unlike on-call, this is bidirectional, however I chose to only allow you to manage schedules with a single rotation (otherwise we wouldn't know which rotation to add/remove people to/from).

### CODE REVIEW

**Review by mike-gregory-ovo — DISMISSED** [review]

File: adapters/opsgenie/schedule/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+func main() {
+	opsgenieConfig := client.Config{
+		ApiKey: "test",
+	}
+	scheduleAdapter, err := schedule.New(&opsgenieConfig, "opsgenie-schedule-id")
+
+	svc := gosync.New(scheduleAdapter)
+
+	// Synchronise an on-call list with something else.
```
Comment (bnjns) [inline adapters/opsgenie/schedule/README.md:38]:
```suggestion
	// Synchronise the participants of a schedule with something else.
```

File: adapters/opsgenie/schedule/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Requirements
+
+You will need to create an [API Key](https://support.atlassian.com/opsgenie/docs/api-key-management/) with the following
+permissions:
+
+| Access rights |
+|:--------------|
+| Read          |
+| Write         |
```
Comment (bnjns) [inline adapters/opsgenie/schedule/README.md:14]:
```suggestion
| Upgrade       |
```

File: adapters/opsgenie/schedule/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+## Requirements
+
+You will need to create an [API Key](https://support.atlassian.com/opsgenie/docs/api-key-management/) with the following
+permissions:
+
+| Access rights |
+|:--------------|
+| Read          |
+| Upgrade       |
```
Comment (bnjns) [inline adapters/opsgenie/schedule/README.md:14]:
```suggestion
| Update        |
```

File: adapters/opsgenie/schedule/schedule.go [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+type Schedule struct {
+	client     iOpsgenieSchedule
+	scheduleID string
+	schedule   *ogSchedule.GetResult
+	logger     *log.Logger
+}
+
+var errMultipleRotations = errors.New("gosync can only manage schedules with a single rotation")
+var errNoRotations = errors.New("gosync cannot create rotations - you must have 1 already defined for schedule")
```
Comment (mike-gregory-ovo) [inline adapters/opsgenie/schedule/schedule.go:36]: Can you export these so that it's easier for downstream users?

**Review by mike-gregory-ovo — APPROVED** [review]
👍

**Pull request merged**